spynnaker/pyNN/extra_algorithms/delay_support_adder.py

```diff
@@ -0,0 +1,261 @@
+        :param int post_vertex_max_delay: delay supported by post vertex.
+        :param int max_delay_needed: the max delay needed by this app edge.
+        :param ApplicationGraph app_graph: the app graph.
+        :return: the DelayExtensionAppVertex
+        """
+
+        # get delay extension vertex if it already exists.
+        delay_app_vertex = self._app_to_delay_map.get(
+            app_outgoing_edge_partition, None)
+        if delay_app_vertex is None:
+            # build delay app vertex
+            delay_name = "{}_delayed".format(app_edge.pre_vertex.label)
+            delay_app_vertex = DelayExtensionVertex(
+                app_edge.pre_vertex.n_atoms, post_vertex_max_delay,
+                max_delay_needed - post_vertex_max_delay, app_edge.pre_vertex,
+                label=delay_name)
+
+            # set trackers
+            delay_app_vertex.splitter = (
+                SplitterDelayVertexSlice(app_edge.pre_vertex.splitter))
+            app_graph.add_vertex(delay_app_vertex)
+            self._app_to_delay_map[app_outgoing_edge_partition] = (
+                delay_app_vertex)
+
+            # build afferent app edge
+            delay_pre_edge = DelayAfferentApplicationEdge(
+                app_edge.pre_vertex, delay_app_vertex,
+                label="{}_to_DelayExtension".format(
+                    app_edge.pre_vertex.label))
+            self._delay_pre_edges.append(delay_pre_edge)
+        return delay_app_vertex
+
+    def _check_delay_values(
+            self, app_edge, user_max_delay, machine_time_step, synapse_infos):
+        """ checks the delay required from the user defined max, the max delay
+            supported by the post vertex splitter and the delay Extensions.
+
+        :param ApplicationEdge app_edge: the undelayed app edge
+        :param int user_max_delay: user max delay of the sim.
+        :param int machine_time_step: machine time step of the sim.
+        :param iterable[SynapseInfo] synapse_infos: iterable of synapse infos
+        :return:tuple of max_delay_needed, post_vertex_max_delay, bool.
+        """
+
+        # get max delay required
+        max_delay_needed = max(
+            synapse_info.synapse_dynamics.get_delay_maximum(
+                synapse_info.connector, synapse_info)
+            for synapse_info in synapse_infos)
+
+        # check max delay works
+        if max_delay_needed > user_max_delay:
+            logger.warning(self.END_USER_MAX_DELAY_DEFILING_ERROR_MESSAGE)
+
+        # get if the post vertex needs a delay extension
+        post_splitter = app_edge.post_vertex.splitter
+        if not isinstance(
+                post_splitter, AbstractSpynnakerSplitterDelay):
+            raise DelayExtensionException(
+                self.INVALID_SPLITTER_FOR_DELAYS_ERROR_MSG.format(
+                    app_edge.post_vertex, post_splitter, app_edge))
+        if not post_splitter.accepts_edges_from_delay_vertex():
```

Comment:
this check is in the wrong place. you need to check if you NEED a delay extension BEFORE doing this check. 

line 245 would be the correct place for this

Reply:
fixed